patches/tModLoader/Terraria.ModLoader.Core/LogArchiver.cs

```diff
@@ -13,8 +13,22 @@ namespace Terraria.ModLoader.Core
 	internal static class LogArchiver
 	{
 		internal static void ArchiveLogs() {
-			foreach (var logFile in Directory.GetFiles(Logging.LogDir, "*.old*"))
-				Archive(logFile, Path.GetFileNameWithoutExtension(logFile));
+			if (!Directory.Exists(Logging.LogArchiveDir)) {
+				string[] existingOldLogs = Directory.GetFiles(Logging.LogDir, "*.zip");
+
+				for (int i = 0; i < existingOldLogs.Length; i++) {
+					try {
+						File.Delete(existingOldLogs[i]);

```

Comment:
Move the existing files to the new directory instead of deleting them. DeleteOldArchives() will handle the old ones.

Reply:
Agreed with above, if you make this change I'll merge it.

Reply:
done